GitHub Workflows caching

LukaDeka opened the pull request:
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
--- Feature


**What is the current behavior?** (You can also link to an open issue here)
--- There is no caching in Github Workflow for Docker and the images get pulled each time.
This causes the rate limit for Docker to be hit, failing the integration tests


**What is the new behavior (if this is a feature change)? If possible add a screenshot.**
--- There is a caching layer


**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
--- No


**Other information**:
---

ClProsser commented on .github/workflows/install-test-onpush.yml:
```diff
@@ -22,11 +23,53 @@ jobs:
+        run: echo "PIPENV_VENV_IN_PROJECT=1" >> $GITHUB_ENV
+
+      - name: Install Pipenv
+        run: pip install --upgrade pipenv
```
You are right, I referenced the wrong code snippet.

I actually wanted to reference this one: 
```yaml
- name: Install dependencies
  run: pipenv install --deploy --ignore-pipfile
```

LukaDeka commented:
The `lint.yaml` Workflow execution time went down from 12m to 6m, I don't know if that's the expected speed. For now, I couldn't figure out how to speed it up even more.

I also added a check, so `.last_checked` is only saved if running on the `main` branch. I have only tested this on this branch by uncommenting that check.

I added caching to `server-tests.yml` as well, which has never run on this repo (because of the `e-m-b-a` owner check. Even after uncommenting that check, this Workflow didn't run.

ClProsser commented on .github/workflows/django-tests.yml:
```diff
@@ -9,19 +9,65 @@ jobs:
         uses: actions/checkout@v3
-      - name: Install python libs
+
+      # - name: Install python libs
+      #  run: |
+      #    sudo apt-get update
+      #    sudo apt-get install -y python3-dev
+      #    sudo pip install --upgrade pipenv
+
```
Please remove this comment

ClProsser commented on .github/workflows/lint.yml:
```diff
@@ -17,43 +18,97 @@ jobs:
+
+      - name: Install dependencies
+        if: steps.pipenv-cache.outputs.cache-hit != 'true'
+        run: pipenv install --deploy --ignore-pipfile --dev # TODO: is the --dev necessary?
```
In my opinion, `--dev` is not required (based on `--ignore-pipfile`) and the `Pipfile.lock` file does not store whether a dependency is a dev dependency or not. 

However, as we are not sure about that, I'd keep `--dev` and just remove the comment.

ClProsser reviewed (changes requested):
Please fix the last two conversations. Great work!

LukaDeka commented:
Thanks for the review! I'll merge this after the tests finish running.
